scripts/pipelines/gcloud/templates/test/test-pipeline.cfg

```
@@ -0,0 +1,39 @@
+#!/bin/bash
+# Mandatory flags.
+mandatoryFlags="$pipelineName,$localDirectory,$language,$buildPipelineName,"
+# Path to the templates.
+templatesPath="scripts/pipelines/gcloud/templates/test"
+# Path to the common templates folder
+commonTemplatesPipelinePath="scripts/pipelines/common/templates/test"
+# YAML file name.
+yamlFile="test-pipeline.yml"
+# Script name.
+scriptFile="test.sh"
+# Source branch.
+sourceBranch="feature/test-pipeline"
+# Description of the trigger
+triggerDescription="Trigger for test pipeline"
+# Branch trigger
+branchTrigger='.*'
+# Previous pipeline
+previousPipelineyaml='build-pipeline.yml'
+
+case $language in
+    node | angular) dockerImage="node" ;;
+    python) dockerImage="python:${languageVersion}-slim" ;;
+    flutter) dockerImage="${registryLocation}-docker.pkg.dev/${gCloudRepo}/flutter/flutter:${languageVersion}" ;;
+    quarkus*) dockerImage="toDefine" ;;
+    *) echo -e "${red}Error: Specified language '${language}' is not supported." >&2; exit 1
+esac
+# Function that copies the script to build the application.
+function copyScript {
+    # Copy the script.
+    cp "${hangarPath}/${commonTemplatesPipelinePath}/${language}-${scriptFile}" "${localDirectory}/${scriptFilePath}/${scriptFile}"
+}
+
+# Function that adds the variables to be used in the pipeline.
+function addPipelineVariables {
+
+    # Setting the list of variable that will be used for this trigger
+    subsitutionVariable="_DOCKER_IMAGE=${dockerImage},_BUILD_PIPELINE_NAME=${buildPipelineName},_BRANCH_NAME="
```

Branch name empty

---

```suggestion
    subsitutionVariable="_DOCKER_IMAGE=${dockerImage},_BUILD_PIPELINE_NAME=${buildPipelineName},_BRANCH_NAME=\$BRANCH_NAME"
```

Better having value by default  that is the built-in var

---

@ultymatom are you sure you can use default pipeline variables inside substitution variables?

---

yes I did it for the deploy